src/main/java/edu/ucla/library/iiif/fester/verticles/V3ManifestVerticle.java

```diff
@@ -414,61 +417,82 @@ private void updatePages(final Message<JsonObject> aMessage) throws JsonProcessi
             final String[] columns = iterator.next();
             final String pageID = columns[aCsvHeaders.getItemArkIndex()];
             final String pageLabel = columns[aCsvHeaders.getTitleIndex()];
+            final Optional<String> format = CsvParser.getMetadata(columns, aCsvHeaders.getMediaFormatIndex());
+
             final String encodedPageID = URLEncoder.encode(pageID, StandardCharsets.UTF_8);
-            final String pageURI = StringUtils.format(SIMPLE_URI, aImageHost, encodedPageID);
+            final String pageURI;
+            final Canvas canvas = new Canvas(aMinter, pageLabel);
+            final int width;
+            final int height;
+            final float duration;
 
-            String resourceURI = StringUtils.format(Constants.SAMPLE_URI_TEMPLATE, pageURI,
-                    Constants.DEFAULT_SAMPLE_SIZE);
-            Canvas canvas;
-            ImageContent image;
+            String resourceURI;
 
-            try {
-                final ImageInfoLookup infoLookup = new ImageInfoLookup(pageURI);
-                final int width = infoLookup.getWidth();
-                final int height = infoLookup.getHeight();
-
-                // Create a canvas using the width and height of the related image
-                canvas = new Canvas(aMinter, pageLabel);
-                image = new ImageContent(resourceURI)
-                        .setServices(new ImageService2(ImageService2.Profile.TWO, pageURI));
-                canvas.setWidthHeight(width, height).paintWith(aMinter, image);
-            } catch (final ImageNotFoundException | IOException details) {
-                LOGGER.info(MessageCodes.MFS_078, pageID);
-
-                if (aPlaceholderImage != null) {
-                    try {
-                        final ImageInfoLookup placeholderLookup = new ImageInfoLookup(aPlaceholderImage);
-                        final int width = placeholderLookup.getWidth();
-                        final int height = placeholderLookup.getHeight();
-                        final int size;
-
-                        if (width >= Constants.DEFAULT_SAMPLE_SIZE) {
-                            size = Constants.DEFAULT_SAMPLE_SIZE;
-                        } else {
-                            size = width;
-                        }
+            // We've already validated the MIME type in CsvParser, so it's fine to just check for a substring here
+            if (format.isPresent() && format.get().contains("video/")) {
+                final VideoContent video;
+
+                resourceURI = CsvParser.getMetadata(columns, aCsvHeaders.getAudioVideoAccessUrlIndex()).get();
+                video = new VideoContent(resourceURI);
+                // TODO: set format on video
```

Comment:
Was leaving this TODO in intentional?

Reply:
I am actually not sure. Currently, `format` is set automatically if the resource id has a filename extension that JP can infer a MimeType from (also, the `setFormat` method is not visible on `VideoContent`). I forgot to remove this comment when I realized this, but now I realize that this behavior might not be desirable (e.g. if the id ends in `.m3u8`).

Reply:
Do you think `setFormat()` should be visible, overriding the default detection?

Reply:
I think it would be good to have the option to use it, in case the URL has no extension or if the extension is incorrect.

Reply:
Would you create a ticket on that project. I don't think it affects us for our first round of materials though. We're controlling the extension (mp4).

Reply:
Okay, great. And [done](https://github.com/ksclarke/jiiify-presentation/issues/94).